File: decompiled/sources/com/hjq/http/config/LogStrategy.java
package com.hjq.http.config;

import android.text.TextUtils;
import android.util.Log;
import com.hjq.http.EasyConfig;
import com.hjq.http.EasyHttp;
import com.hjq.http.config.ILogStrategy;

public final class LogStrategy implements ILogStrategy {
    public /* synthetic */ void print() {
        ILogStrategy.CC.$default$print(this);
    }

    public void print(String str) {
        String logTag = EasyConfig.getInstance().getLogTag();
        if (str == null) {
            str = "null";
        }
        Log.i(logTag, str);
    }

    public void json(String str) {
        String formatJson = ILogStrategy.CC.formatJson(str);
        if (!TextUtils.isEmpty(formatJson)) {
            String str2 = " \n" + formatJson;
            if (((long) str2.length()) <= ((long) 3072)) {
                print(str2);
                return;
            }
            while (str2.length() > 3072) {
                String substring = str2.substring(0, 3072);
                str2 = str2.replace(substring, "");
                print(substring);
            }
            print(str2);
        }
    }

    public void print(String str, String str2) {
        print(str + " = " + str2);
    }

    public void print(Throwable th) {
        Log.e(EasyConfig.getInstance().getLogTag(), th.getMessage(), th);
    }

    public void print(StackTraceElement[] stackTraceElementArr) {
        int length = stackTraceElementArr.length;
        int i = 0;
        while (i < length) {
            StackTraceElement stackTraceElement = stackTraceElementArr[i];
            int lineNumber = stackTraceElement.getLineNumber();
            String className = stackTraceElement.getClassName();
            if (lineNumber <= 0 || className.startsWith(EasyHttp.class.getPackage().getName())) {
                i++;
            } else {
                print("RequestCode = (" + stackTraceElement.getFileName() + ":" + lineNumber + ") ");
                return;
            }
        }
    }
}

File: decompiled/sources/com/hjq/http/config/LogStrategy.kt
package com.hjq.http.config

import android.text.TextUtils
import android.util.Log
import com.hjq.http.EasyConfig
import com.hjq.http.EasyHttp

class LogStrategy : ILogStrategy {
    override fun print(str: String?) {
        val logTag = EasyConfig.getInstance().logTag
        Log.i(logTag, str ?: "null")
    }

    override fun json(str: String?) {
        val formatJson = ILogStrategy.formatJson(str)
        if (!TextUtils.isEmpty(formatJson)) {
            var message = " \n$formatJson"
            if (message.length.toLong() <= 3072L) {
                print(message)
                return
            }
            while (message.length > 3072) {
                val substring = message.substring(0, 3072)
                message = message.replace(substring, "")
                print(substring)
            }
            print(message)
        }
    }

    override fun print(str: String?, str2: String?) {
        print(str + " = " + str2)
    }

    override fun print(th: Throwable) {
        Log.e(EasyConfig.getInstance().logTag, th.message, th)
    }

    override fun print(stackTraceElementArr: Array<StackTraceElement>) {
        for (stackTraceElement in stackTraceElementArr) {
            val lineNumber = stackTraceElement.lineNumber
            val className = stackTraceElement.className
            if (lineNumber > 0 && !className.startsWith(EasyHttp::class.java.`package`.name)) {
                print("RequestCode = (" + stackTraceElement.fileName + ":" + lineNumber + ") ")
                return
            }
        }
    }
}
